src/web/builds.rs

```diff
@@ -37,11 +38,30 @@ impl_webpage! {
 pub fn build_list_handler(req: &mut Request) -> IronResult<Response> {
     let router = extension!(req, Router);
     let name = cexpect!(req, router.find("name"));
-    let version = cexpect!(req, router.find("version"));
+    let req_version = router.find("version");
 
     let mut conn = extension!(req, Pool).get()?;
     let limits = ctry!(req, Limits::for_crate(&mut conn, name));
 
+    let version =
+        match match_version(&mut conn, name, req_version).and_then(|m| m.assume_exact())? {
+            MatchSemver::Exact((version, _)) => version,
+
+            MatchSemver::Semver((version, _)) => {
+                let url = ctry!(
+                    req,
+                    Url::parse(&format!(
+                        "{}/crate/{}/{}/builds",
+                        redirect_base(req),
+                        name,
+                        version
+                    )),
+                );
+
+                return Ok(super::redirect(url));
+            }
+        };
```

Comment:
This is repeated a few places in the code base now. At some point I'd like to reduce the duplication, but it doesn't need to block this PR.